### ISSUE TITLE
[NEWS] Nowe zasoby - Zarządzaj tagami i załącznikami w ofertach

### ISSUE THREAD

**Issue by MarcinLuczak** [allegro/allegro-api#646]
Udostępniliśmy zestaw zasobów do zarządzania tagami w ofertach oraz do dodawania załączników. Uzupełniliśmy także o tagi i załączniki nasz poradnik [“Jak wystawić ofertę w Allegro REST API”](https://developer.allegro.pl/sale/).

Przygotowaliśmy następujące metody:

- [GET /sale/offer-tags?user.id={user_ID}](https://developer.allegro.pl/news/2018-09-24-Tagi-zalaczniki/#PobierzTagi) - chcę pobrać wszystkie tagi, które są dodane na moim koncie.
- [POST /sale/offer-tags](https://developer.allegro.pl/news/2018-09-24-Tagi-zalaczniki/#DodajDoKonta) - chcę dodać tagi do mojego konta.
- [DELETE /sale/offer-tags/{tag_ID}](https://developer.allegro.pl/news/2018-09-24-Tagi-zalaczniki/#UsunTagZKonta) - chcę usunąć tag z mojego konta.
- [PUT /sale/offer-tags/{tag_ID}](https://developer.allegro.pl/news/2018-09-24-Tagi-zalaczniki/#ZmienNazwe) - chcę zmienić nazwę taga na moim koncie.
- [GET /sale/offers/{offer_ID}/tags](https://developer.allegro.pl/news/2018-09-24-Tagi-zalaczniki/#PobierzTagiZOferty) - chcę pobrać tagi, które są przypisane do mojej oferty.
- [POST /sale/offers/{offer_ID}/tags](https://developer.allegro.pl/news/2018-09-24-Tagi-zalaczniki/#PrzypiszTagiDoOferty) - chcę przypisać tagi do mojej oferty.
- [POST /sale/offer-attachments](https://developer.allegro.pl/sale/#Stworz) - chcę dodać załącznik.

Więcej informacji znajdziecie w [artykule](https://developer.allegro.pl/news/2018-09-24-Tagi-zalaczniki/).

**Comment by kfijalko** [comment]
W jaki sposób mogę ustawić kolejność wyświetlania tagów?

**Comment by MarcinLuczak** [comment]
@kfijalko w zakładce [Tagi w Moim Allegro](https://allegro.pl/offerShopsTagsFrontend/myaccountTags/). Zasoby REST API tego nie obsługują.

**Comment by TimothyKoval** [comment]
Oczywiście wszystko pod górę... Czemu najpierw muszę "tworzyć obiekt załącznika" a później dopiero przesyłać plik? Nie można tego zrobić w jednym requeście?

**Comment by kfijalko** [comment]
@MarcinLuczak dopiszcie gdzieś w planach rozwoju mozliwość sortowania w rest api. Mam kilka tysięcy tagów i ich ręczne sortowanie nie wchodzi w grę.

**Comment by MarcinLuczak** [comment]
@TimothyKoval Zdecydowaliśmy się na takie rozwiązanie, ponieważ jest bardziej przejrzyste i elastyczne. Oczywiście istnieją rozwiązania, które umożliwiają dodanie załącznika w jednym requeście, ale są bardziej skomplikowane i nie do końca zgodne ze sztuką. Obecne rozwiązanie jest rozwiązaniem optymalnym. Nie planujemy jego zmiany.

@kfijalko dzięki, przekazałem tą sugestię do odpowiedniego działu.

**Comment by bugas23** [comment]
Czy dodawanie tagów możliwe jest jedynie do ofert aktywnych?
Jeżeli tak to trochę to komplikuje wystawianie ofert.

**Comment by MarcinLuczak** [comment]
@bugas23 tak, tagi możesz przypisać tylko do aktywnej oferty. Poprzez WWW również możesz przypisać tagi tylko do aktywnych ofert.

**Comment by bugas23** [comment]
@MarcinLuczak No nie - na www mogę przypisać tagi na etapie wystawiana oferty (czyli de facto do draftu). Później oferta jest aktywowana już z przypisanymi tagami. Myślę, że tak to powinno wyglądać również w rest api. 

Inaczej muszę robić skrypt sprawdzający czy oferta jest aktywna, aby przypisać do nich oddzielne tagi - nieuzasadnione komplikacje i mnożenie ilości zapytań.

**Comment by MarcinLuczak** [comment]
@bugas23 to trochę inna sytuacja, która nie do końca się przekłada na drafty oferty, które są dostępne przez REST API, ale faktycznie mogłem to trochę lepiej wytłumaczyć.

Obecnie nie mamy w planach umożliwienia dodawania tagów do nieaktywnej oferty, ale dziękuję za sugestię. Przekażę ją do odpowiedniego działu.

**Comment by bugas23** [comment]
@MarcinLuczak - rozumiem, że rozmawiamy w dużym uproszczeniu, ale chyba tylko tak w miarę szybko udało nam się porozumieć o jaką konkretnie sytuację nam chodzi.

Jeżeli nie dodawania tagów do nieaktywnych ofert to proszę przemyśleć inne narzędzie, które umożliwi obejście tego problemu.

Proszę przemnożyć ile zapytań wygeneruje skrypt sprawdzający co np. 1 sek. przez godzinę czy np. 100 ofert jest aktywnych... i jeszcze przez ilość użytkowników. Na razie tagi nie są tak popularne, ale kiedy ludzie zrozumieją ich działanie to będzie problem.

Dziękuję za poświęcony czas i pozdrawiam.

**Comment by MarcinLuczak** [comment]
> rozumiem, że rozmawiamy w dużym uproszczeniu, ale chyba tylko tak w miarę szybko udało nam się porozumieć o jaką konkretnie sytuację nam chodzi.

Tak, jak najbardziej. Również dziękuję.
